### PR TITLE
Jd/kwargs implementation

### PR DESCRIPTION
**Pull request by jd-lara**
Closes PSIMA-60.

### CODE REVIEW

File: src/core/psi_container.jl [hunk 1]
```diff
@@ -157,41 +198,35 @@ mutable struct PSIContainer
...
-            initial_time,
+            settings,
             variables,
             constraints,
             cost_function,
             expressions,
             parameters,
             initial_conditions,
             pm,
+            true,
```
Comment (daniel-thom) [inline src/core/psi_container.jl]: Can this ever be false?

File: src/core/simulation_stages.jl [hunk 2]
```diff
@@ -37,61 +38,130 @@ end
...
-        use_parameters = true,
-        initial_time = initial_time,
-        horizon = horizon,
-    )
-    _build!(stage.internal.psi_container, stage.template, stage.sys; kwargs...)
-    stage.internal.warm_start_enabled = MOI.supports(
+    stage_instantiated(stage) && reset!(stage)
+    psi_container = get_psi_container(stage)
+    psi_container.settings.horizon = horizon
+    psi_container.settings.initial_time = initial_time
```
Comment (daniel-thom) [inline src/core/simulation_stages.jl]: Seems strange that you override these two parameters here.

File: src/core/simulation_stages.jl [hunk 3]
```diff
@@ -37,61 +38,130 @@ end
...
         MOI.VariableIndex,
     )
+    if !solver_supports_warm_start
+        solver_name = JuMP.solver_name(psi_container.JuMPmodel)
+        @warn("$(solver_name) in stage $(stage.internal.number) does not support warm start")
+    end
+    psi_container.settings.use_warm_start = solver_supports_warm_start
     stage_resolution = PSY.get_forecasts_resolution(stage.sys)
     stage.internal.end_of_interval_step = Int(stage_interval / stage_resolution)
+    stage.internal.instantiated = true
```
Comment (daniel-thom) [inline src/core/simulation_stages.jl]: `is_built`?

File: src/core/psi_container.jl (hunk 1 above)
Comment (jd-lara) [reply src/core/psi_container.jl]: Ok, I will remove these

File: src/core/simulation_stages.jl (hunk 3 above)
Comment (jd-lara) [reply src/core/simulation_stages.jl]: ok, I will rename

File: src/core/operations_problem.jl [hunk 4]
```diff
@@ -40,139 +38,159 @@ end
...
                     ::Type{T},
-                    sys::PSY.System;
+                    sys::PSY.System,
+                    jump_model::Union{Nothing, JuMP.AbstractModel}=nothing;
                     kwargs...) where {M<:AbstractOperationsProblem,
-                                    T<:PM.AbstractPowerFormulation}
-This uses the Abstract Power Formulation to build the model reference and
-the optimization model and populates the operation model struct.
+                                      T<:PM.AbstractPowerFormulation}
+This Return an unbuilt operation problem of type M with the specific system and network model T.
```
Comment (daniel-thom) [inline src/core/operations_problem.jl]: Delete `This`

File: src/core/operations_problem.jl [hunk 5]
```diff
@@ -40,139 +38,159 @@ end
...
     OperationsProblem(::Type{T},
-                    sys::PSY.System;
+                    sys::PSY.System,
+                    jump_model::Union{Nothing, JuMP.AbstractModel}=nothing;
                     kwargs...) where {M<:AbstractOperationsProblem,
                                       T<:PM.AbstractPowerFormulation}
-This uses the Abstract Power Formulation to build the model reference and
-the optimization model and populates the operation model struct.
-***Note:*** the abstract operation model is set to the default operation model
+This return an unbuilt operation problem of type GenericOpProblem with the specific system and network model T.
```
Comment (daniel-thom) [inline src/core/operations_problem.jl]: Delete `This`.  `Return`

**Pull request merged**